scipy/optimize/_differentialevolution.py

```diff
@@ -887,7 +949,15 @@ def solve(self):
             message=status_message,
             success=(warning_flag is not True))
 
-        if self.polish:
+        if self.polish and not np.all(self.integrality):
+            # can't polish if all the parameters are integers
+            if np.any(self.integrality):
+                # set the lower/upper bounds equal so that any integrality
+                # constraints work.
+                limits, integrality = self.limits, self.integrality
```

Comment:
This is just to save line length below, right? Looks like `limits` and `integrality` are not used anywhere otherwise, but you're mutating the same array object, so that's OK.

Reply:
Yep, saving line length